File: Python/数组/167 Two Sum II - Input array is sorted.py
def twoSum(numbers, target):
    n = len(numbers)
    low = 0
    high = n - 1
    while numbers[low] + numbers[high] != target:
        s = numbers[low] + numbers[high]
        if s < target:
            low += 1
        if s > target:
            high -= 1
    return [low + 1, high + 1]


"""
[3,24,50,79,88,150,345] 
200
"""
print(twoSum([3,24,50,79,88,150,345], 200))

File: JS/数组Array/167 Two Sum II - Input array is sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(numbers, target) {
    let low = 0;
    let high = numbers.length-1;
    let s = numbers[low] + numbers[high];
    console.log(s);
    while (s != target) {
        if (s > target) {
            high--;
        }
        if (s < target) {
            low++;
        }
        s = numbers[low] + numbers[high];
    }
    return [low+1, high+1];
     
};

console.log(twoSum([2,7,11,15], target = 9)); 